geniza/corpus/management/commands/export_metadata.py

```diff
@@ -1,30 +1,189 @@
+            self.path_footnotes_csv,
+            self.path_sources_csv,
+            self.path_fragments_csv,
+        ]
+
+    def add_and_commit(self):
+        for fn in self.paths:
+            if os.path.exists(fn):
+                self.repo.index.add(fn)
+        self.repo.index.commit(
+            "Auto-syncing @ " + timezone.now().strftime("%Y-%m-%d %H:%M:%S")
+        )
+
+    def write_local(self):
+        # write docs
+
+        with self.timer("Exporting document objects"):
+            PublicDocumentExporter(progress=self.progress).write_export_data_csv(
+                self.path_documents_csv
+            )
+
+        # write sources
+        with self.timer("Exporting source objects"):
+            SourceExporter(progress=self.progress).write_export_data_csv(
+                self.path_sources_csv
+            )
+
+        # write footnotes
+        with self.timer("Exporting footnote objects"):
+            FootnoteExporter(progress=self.progress).write_export_data_csv(
+                self.path_footnotes_csv
+            )
+
+    def sync_remote(self):
+        """Sync local repository content with origin repository. Assumes
+        :meth:`setup_repo` has already been run, and any new or modified
+        files have been committed."""
+        try:
+            with self.timer("Pulling repository"):
+                origin = self.repo.remote(name="origin")
+                # pull any remote changes since our last commit
+                origin.pull()
+
+            # commit?
+            with self.timer("Adding any changes"):
+                self.add_and_commit()
+
+            # push data updates
+            with self.timer("Pushing any changes"):
+                result = origin.push()
+                if result:
+                    info = result[0]
+                    self.print(f"Result = {info.flags}")
+        except ValueError:
+            self.print("No origin repository, unable to push updates")
+
+
+class Command(BaseCommand, Timerable):
     def print(self, *x, **y):
         """
         A stdout-friendly method of printing for manage.py commands
         """
-        self.stdout.write(" ".join(str(xx) for xx in x), ending="\n", **y)
+        if not hasattr(self, "to_print") or self.to_print:
+            self.stdout.write(" ".join(str(xx) for xx in x), ending=y.get("end", "\n"))
 
     def add_arguments(self, parser):
-        ofn = f'pgp_documents-{timezone.now().strftime("%Y%m%dT%H%M%S")}.csv'
-        parser.add_argument(
-            "-o", "--output_filename", type=str, default=ofn, required=False
-        )
+        parser.add_argument("-w", "--write", action="store_true")
+        parser.add_argument("-s", "--sync", action="store_true")
+
+        parser.add_argument("-p", "--path", type=str, default="", required=False)
+        parser.add_argument("-u", "--url", type=str, default="", required=False)
+
+        parser.add_argument("-q", "--quiet", action="count", default=0)
```

Comment:
django manage commands already have a verbosity argument that we inherit, would be better to use that one for consistency

https://docs.djangoproject.com/en/3.2/ref/django-admin/#cmdoption-verbosity